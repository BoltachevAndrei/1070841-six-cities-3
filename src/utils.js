import {SortTypeList} from './types';

export const capitalizeString = (str) => str[0].toUpperCase() + str.slice(1);

export const extend = (object1, object2) => Object.assign({}, object1, object2);

export const getCities = (data) => Array.from(new Set(data.map((element) => element.city)).values());

export const getOffersByCity = (offers, city) => offers.filter((offer) => offer.city === city);

export const sortOffers = (offers, sortType) => {
  switch (sortType) {
    case SortTypeList.HIGH_TO_LOW:
      return offers.sort((prev, next) => next.price - prev.price);
    case SortTypeList.LOW_TO_HIGH:
      return offers.sort((prev, next) => prev.price - next.price);
    case SortTypeList.TOP_RATED_FIRST:
      return offers.sort((prev, next) => next.rating - prev.rating);
  }
  return offers;
};

export const getCitiesList = (offers) => {
  const citiesNames = Array.from(new Set(offers.map((element) => element.city.name)).values());
  const citiesData = Array.from(offers.values()).map((offer) => offer.city);
  const result = citiesNames.map((name) => citiesData.filter((city) => city.name === name)[0]);
  return result;
};

export const doNothing = () => {
  // do nothing
};
