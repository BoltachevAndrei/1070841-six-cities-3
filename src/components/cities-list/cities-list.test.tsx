import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CitiesList from './cities-list';
import {doNothing} from '../../utils';

const citiesList = [
  {
    name: `Test city`,
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10
    }
  }
];

const activeCity = {
  name: `Test city`,
  location: {
    latitude: 52.370216,
    longitude: 4.895168,
    zoom: 10
  }
};

it(`CitiesList renders correctly`, () => {
  const tree = renderer
    .create(<CitiesList
      citiesList={citiesList}
      activeCity={activeCity}
      onCityClick={doNothing}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
