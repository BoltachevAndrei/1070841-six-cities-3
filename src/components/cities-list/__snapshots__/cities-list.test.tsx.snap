// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList renders correctly 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
    onClick={[Function]}
  >
    <a
      className="locations__item-link tabs__item tabs__item--active"
    >
      <span>
        Test city
      </span>
    </a>
  </li>
</ul>
`;
